packages/back-end/src/controllers/features.ts

```diff
@@ -215,11 +220,18 @@ export async function getFeaturesPublic(req: Request, res: Response) {
       );
     }
 
+    const environmentDoc = context.org?.settings?.environments?.find(
+      (e) => e.id === environment
+    );
+    const filteredProjects = environmentDoc
+      ? filterProjectsByEnvironment(projects, environmentDoc, true)
+      : projects;
+
     const defs = await getFeatureDefinitions({
       context,
       capabilities,
       environment,
-      projects,
+      projects: filteredProjects,
```

Comment:
Each of the "get SDK payload" type of endpoints need to filter the definitions by the whitelist of projects (the filtering is not applied to the cached payload and is JIT). Previously a project value of `[]` meant "all projects: don't filter anything". But now with environments having their own set of whitelist projects, we need to make sure we're passing in the filtered subset of projects defined in both the SDK Connection and Environment, especially since `[]` could mean "filter using environment's projects".